src/app/card/cardList.component.ts

```diff
@@ -1,57 +1,35 @@
-import { ChangeDetectionStrategy, Component, DestroyRef, OnInit } from '@angular/core';
-import { Card, Product } from '../interfaces';
-import { CommonModule } from '@angular/common';
-import { Observable } from 'rxjs';
-import { ProductService } from '../Services/product.service';
-import { CardService } from '../Services/card.service';
-import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import {ChangeDetectionStrategy, Component, DestroyRef, inject} from '@angular/core';
+import {Card} from '../interfaces';
+import {CommonModule} from '@angular/common';
+import {ProductService} from '../Services/product.service';
+import {CardService} from '../Services/card.service';
 
 @Component({
-  selector: 'app-card',
-  templateUrl: './cardList.component.html',
-  styleUrl: './cardList.component.css',
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  standalone: true,
-  imports: [CommonModule],
+    selector: 'app-card',
+    templateUrl: './cardList.component.html',
+    styleUrl: './cardList.component.css',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    standalone: true,
+    imports: [CommonModule],
 })
-export class CardListComponent implements OnInit {
-  private _products$: Observable<Product[]>;
-  protected _cardList$: Observable<Card[]>;
+export class CardListComponent {
 
-  protected _products: Product[];
+    private readonly productService = inject(ProductService)
+    private readonly cardService = inject(CardService)
```

Comment:
```suggestion
    private readonly _productService = inject(ProductService)
    private readonly _cardService = inject(CardService)
```

C'est aussi valable pour les services